Lib/fontTools/cffLib.py

```diff
@@ -27,21 +32,38 @@ def __init__(self):
 		pass
 
 	def decompile(self, file, otFont):
-		sstruct.unpack(cffHeaderFormat, file.read(4), self)
-		assert self.major == 1 and self.minor == 0, \
-				"unknown CFF format: %d.%d" % (self.major, self.minor)
-
-		file.seek(self.hdrSize)
-		self.fontNames = list(Index(file))
-		self.topDictIndex = TopDictIndex(file)
-		self.strings = IndexedStrings(file)
-		self.GlobalSubrs = GlobalSubrsIndex(file)
-		self.topDictIndex.strings = self.strings
-		self.topDictIndex.GlobalSubrs = self.GlobalSubrs
+		global isCFF2
```

Comment:
I have thought about how to do this. If this code needs to be thread safe and re-entrant, I can certainly make the changes, but would rather do so after the current logic has proven stable,